standalone-examples/src/main/java/io/pravega/example/binary/BinaryReader.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.binary;
+
+import io.pravega.client.ClientFactory;
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReaderGroupConfig;
+import io.pravega.client.stream.ReinitializationRequiredException;
+import io.pravega.client.stream.Sequence;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+public class BinaryReader implements Runnable {
```

Comment:
A few suggestions to make this example code more broadly useful:
1. Make it generic by taking the `Serializer<T>` and `Consumer<T>`, then call it `SimpleReader` or something similar.
2. Make it possible to stop the reader with a volatile `isRunning` field and corresponding `stop` method.

Reply:
These are both good suggestions and make this sample more applicable to other uses. Will make these changes.